src/auth/auth.proto

```diff
@@ -157,26 +113,56 @@ message WhoAmIRequest {}
 
 message WhoAmIResponse {
   string username = 1;
-  bool is_admin = 2;
   int64 ttl = 3 [(gogoproto.customname) = "TTL"];
-  ClusterRoles cluster_roles = 4;
 }
 
 //// Authorization data structures
 
-// Scope (actually a "role" in canonical security nomenclature) represents a
-// rough level of access that a principal has to a repo
-enum Scope {
-  // To remove a user's scope from a repo, set their scope to NONE
-  NONE = 0;
-  READER = 1;
-  WRITER = 2;
-  OWNER = 3;
+// Roles represents the set of roles a principal has
+message Roles {
+  map<string, bool> roles = 1;
 }
 
-message ACL {
-  // principal -> scope. All principal names include the structured prefix indicating their type.
-  map<string, Scope> entries = 1;
+// RoleBinding represents the set of roles principals have on a given Resource
+message RoleBinding {
+  // principal -> roles. All principal names include the structured prefix indicating their type.
+  map<string, Roles> entries = 1;
+}
+
+// Permission represents the ability to perform a given operation on a Resource 
+enum Permission {
+  UNKNOWN = 0;
+
+  // CLUSTER_ADMIN encompasses all "admin" APIs, which are generally cluster configuration
+  CLUSTER_ADMIN        = 100;
+
+  REPO_READ            = 200;
+  REPO_WRITE           = 201;
+  REPO_MODIFY_BINDINGS = 202;
+  REPO_DELETE          = 203;
+  REPO_INSPECT_COMMIT  = 204;
+  REPO_LIST_COMMIT     = 205;
+  REPO_DELETE_COMMIT   = 206;
+  REPO_CREATE_BRANCH   = 207;
+  REPO_LIST_BRANCH     = 208;
+  REPO_DELETE_BRANCH   = 209;
+  REPO_INSPECT_FILE    = 210;
+  REPO_LIST_FILE       = 211;
```

Comment:
This is a first pass at splitting out individual permissions from the PFS roles. I suspect we'll need to do a couple revisions to get to something useful.